lib/actors/actor/invocation_scheduler.ex

```diff
@@ -0,0 +1,83 @@
+defmodule Actors.Actor.InvocationScheduler do
+  use GenServer, restart: :transient
+  use Retry
+
+  require Logger
+
+  alias Actors.Registry.ActorRegistry
+  alias Eigr.Functions.Protocol.InvocationRequest
+
+  @impl true
+  def init(_arg) do
+    stored_invocations = ActorRegistry.get_all_invocations()
+
+    Enum.each(stored_invocations, &call_invoke/1)
+
+    {:ok, %{}}
+  end
+
+  @impl true
+  def handle_info({:invoke, request}, state) do
+    decoded_request = InvocationRequest.decode(request)
+
+    retry with: exponential_backoff() |> randomize |> expiry(10_000),
+          atoms: [:error, :exit, :noproc, :erpc, :noconnection],
+          rescue_only: [ErlangError] do
+      ActorRegistry.remove_invocation_request(decoded_request.actor.id.name, request)
+    after
+      _ ->
+        spawn(fn ->
+          Actors.invoke(%{decoded_request | scheduled_to: nil, async: true})
```

Comment:
The "| scheduled_to: nil, async: true" could be done already when you registered the scheduler steps back. Here you would only call invoke with the request retrieved already cleaned for the call